### PR TITLE
Enoch/collision events

### PR DESCRIPTION
**Pull request by enochtsang**
If you drive a car into another car, it'll play a sick sound.

I tried making what you collide into identifiable but it kicked my ass pretty hard.

### CODE REVIEW

File: src/CollisionEventsSubsystem.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+) {
+    // do nothing
+}
+
+void CollisionEventsSubsystem::onAdvance(
+    const physx::PxRigidBody* const* bodyBuffer,
+    const physx::PxTransform* poseBuffer,
+    const physx::PxU32 count
+) {
+    // do nothing
```
Comment (ammunoz) [inline src/CollisionEventsSubsystem.cpp]: It'll be worth a look into as to how we're gonna handle the different collision events between different kinds of objects (e.g., collision between car and wall, etc.). I can imagine these growing into large if branches otherwise.

File: src/PhysicsSystem.cpp [hunk 2]
```diff
@@ -71,6 +74,16 @@ PhysicsSystem::~PhysicsSystem() {
...
+            Event(
+                EventType::VEHICLE_COLLISION,
+                "vehicle_a", collision.first,
+                "vehicle_b", collision.second
+            )
+        );
+    }
+
     const int SIM_STEPS = 4;
     const float TIME_PER_UPDATE = 0.16f;
```
Comment (ammunoz) [inline src/PhysicsSystem.cpp]: I know that it's not part of the PR but shouldn't these go into the anon namespace?

File: src/CollisionEventsSubsystem.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+void CollisionEventsSubsystem::onContact(
+    const physx::PxContactPairHeader& pairHeader,
+    const physx::PxContactPair* pairs,
+    physx::PxU32 nbPairs
+) {
+    collisions_.emplace_back(0, 0);
```
Comment (ammunoz) [inline src/CollisionEventsSubsystem.cpp]: I'm guessing the two 0's here should be representing the object ids involved in the collision? If so, can you please just comment it for now.

File: src/PhysicsSystem.cpp (hunk 2 above)
Comment (enochtsang) [reply src/PhysicsSystem.cpp]: I don't think so, because the scope of these variables is only useful for the update function.

File: src/CollisionEventsSubsystem.cpp (hunk 3 above)
Comment (enochtsang) [reply src/CollisionEventsSubsystem.cpp]: Yea, it should be. But I had a really hard time getting the object ids from the given variables in this callback.

File: src/CollisionEventsSubsystem.cpp (hunk 1 above)
Comment (enochtsang) [reply src/CollisionEventsSubsystem.cpp]: Yea, I think we will not have enough cases that we will care about so I don't think we need any fancy way to handle this. But we shall see, we can overcome that hurdle when we get there.
